src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -147,6 +148,11 @@ private function populateAvailableType($id, Definition $definition)
             $this->types[$type] = $id;
         }
 
+        // Cannot use reflection if the class isn't set
```

Comment:
I've done that on purpose. Associating a service with a type using the `autowiring_types` key doesn't require reflection, so it doesn't require the `class` key to exist.

It allows to do things like associating types with a service definition where the class is associated in a compiler pass executed after the autowiring.